coral-spark/src/main/java/com/linkedin/coral/spark/CoralToSparkSqlCallTransformationUtils.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2022 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.spark;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.calcite.sql.JoinConditionType;
+import org.apache.calcite.sql.JoinType;
+import org.apache.calcite.sql.SqlBasicCall;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlJoin;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlLiteral;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+import com.linkedin.coral.hive.hive2rel.functions.HivePosExplodeOperator;
+import com.linkedin.coral.spark.functions.SqlEmptyOperator;
+import com.linkedin.coral.spark.functions.SqlLateralJoin;
+import com.linkedin.coral.spark.functions.SqlLateralViewAsOperator;
+
+
+/**
+ * Util class to obtain a spark compatible transformed SqlCall.
+ * The transformation may involve change in operator, reordering the operands
+ * or even re-construction of the SqlCall.
+ */
+public class CoralToSparkSqlCallTransformationUtils {
+  private CoralToSparkSqlCallTransformationUtils() {
+  }
+
+  public static SqlCall getTransformedSqlCall(SqlCall sqlCall) {
+    switch (sqlCall.getOperator().kind) {
+      case JOIN:
+        return getTransformedJoinSqlCall(sqlCall);
+      case AS:
+        return getTransformedAsSqlCall(sqlCall);
+      case LATERAL:
+      case COLLECTION_TABLE:
+        return getEmptyOperatorSqlCall(sqlCall);
+      case UNNEST:
+        return getTransformedUnnestSqlCall(sqlCall);
```

Comment:
I think it's a little bit hard to understand if we categorize them by operate kinds.
It would be easer to understand if we just create some methods to do specific conversions like operand reordering for POSEXPLODE and outer join handling.
And it would be great to add comments to these converter methods.

Reply:
Discussed the current class layout offline. If there's an operand reordering for POSEXPLODE, it must be done in its parent SqlCall. I added more comments in the class. Let me know if you feel some methods could be elaborated further.